Make feature tests more stable

cbliard opened the pull request:
Following advices on https://github.com/teamcapybara/capybara/issues/2796

This occurs sometimes that some tests are failing in headless mode: the DOM is not correctly reflecting the state of the page until a screenshot is taken.

This happened in ./spec/features/work_packages/table/queries/query_menu_refresh_spec.rb:44 in job https://github.com/opf/openproject/actions/runs/14965572620/job/42034828494

The screenshot shows that the query menu item is here, but before the screenshot is taken, capybara cannot find it.

This is a shot in the dark. I have no idea if that will really fix the issue mentioned above.

cbliard committed:
Make feature tests more stable

Following advices on https://github.com/teamcapybara/capybara/issues/2796

This occurs sometimes that some tests are failing in headless mode: the
DOM is not correctly reflecting the state of the page until a screenshot
is taken.

This happened in ./spec/features/work_packages/table/queries/query_menu_refresh_spec.rb:44
in job https://github.com/opf/openproject/actions/runs/14965572620/job/42034828494

The screenshot shows that the query menu item is here, but before the
screenshot is taken, capybara cannot find it.

This is a shot in the dark. I have no idea if that will really fix the issue mentioned above.

NobodysNightmare reviewed (approved):
I don't know whether this will fix the issue it wants to fix, but the settings added are well-explained and it's plausible to set them.

Let's try :+1:

cbliard committed:
Fix flaky spec

spec is modules/meeting/spec/features/meeting_backlogs_spec.rb:128
failing job is https://github.com/opf/openproject/actions/runs/14969474602/job/42047670034

Problem is that the menu can disappear before the action is clicked. It
disappears when the message "The meeting has been updated. Reload?" is
displayed. Sometimes it is just not displayed and I could not figure out
why. The action seems to be swallowed.

Fix is to retry.

It's not perfect yet. It can still fail at
`modules/meeting/spec/features/meeting_backlogs_spec.rb:201` when
calling `clear_backlog` to close the modal. The modal is not there.

I'm not sure how to fix this one.
